### PR TITLE
Add support for repos with no tags

### LINKED ISSUE #23: Support repositories that do not yet have any tags

### PR DESCRIPTION
**Pull request by edthedev**
- Add unit tests wrapping fetching current release
- Add unit tests wrapping generation of next release
- Clear rtd-sphinx-theme and docutils error message

Fixes #23

### CODE REVIEW

File: src/farmit/__init__.py [hunk 1]
```diff
@@ -245,10 +246,26 @@ def update_changelog(args: Namespace, changelog_path: str, entry: str):
...
+def get_current_release(repo: Repo):
+    """Get the current release based on last tag."""
+
+    # Check for any tags
+    if not repo.tags:
+        return None
+
+    next_version, _ = get_version().split('.dev')
+    next = Version(next_version)
+    # import pdb;pdb.set_trace()
```
Comment (ddriddle) [inline src/farmit/__init__.py:258]: Remove debug comments:

```suggestion
```

File: src/farmit/__init__.py [hunk 2]
```diff
@@ -245,10 +246,26 @@ def update_changelog(args: Namespace, changelog_path: str, entry: str):
...
 
+def get_current_release(repo: Repo):
+    """Get the current release based on last tag."""
+
+    # Check for any tags
+    if not repo.tags:
+        return None
+
+    next_version, _ = get_version().split('.dev')
+    next = Version(next_version)
```
Comment (ddriddle) [inline src/farmit/__init__.py:257]: Could be a one-liner:

```suggestion
    next = Version(get_version().split('.dev'))  # Version returns next micro release
```

File: tests/test_farmit.py [hunk 3]
```diff
@@ -9,7 +9,6 @@
 
 from git import Repo
 
-
```
Comment (ddriddle) [inline tests/test_farmit.py:12]: Add the whitespace back

```suggestion

```

File: tests/test_farmit.py [hunk 4]
```diff
@@ -56,6 +55,27 @@ def repo(tmp_path, remote_url, request):
     os.chdir(request.config.invocation_dir)
 
 
+@pytest.fixture
+def no_tags_repo(tmp_path, remote_url, request):
```
Comment (ddriddle) [inline tests/test_farmit.py:59]: Please make this code DRY. `repo` should call `no_tags_repo`.

File: tests/test_farmit.py [hunk 5]
```diff
@@ -150,3 +170,20 @@ def test_version_increase(repo, capsys):
...
         "Release 1.0.11\n\n+ 1st commit\n"
+
+@pytest.mark.parametrize("version, expected, args", [
+    ('1.1.0', '1.1.1', ['micro', '--dry-run']),
+    ('1.1.0', '1.2.0', ['minor', '--dry-run']),
+    ('1.1.0', '2.0.0', ['major', '--dry-run']),
+    ('1.1.0', '9.8.9', ['9.8.9', '--dry-run']),
+])
+def test_get_next_release(version, expected, args):
+
```
Comment (ddriddle) [inline tests/test_farmit.py:181]: Remove whitespace:

```suggestion
```

File: tests/test_farmit.py [hunk 6]
```diff
@@ -150,3 +170,20 @@ def test_version_increase(repo, capsys):
...
+    parsed_args = farmit.init_parser(farmit._easter_egg()).parse_args(args)
+    next = farmit.get_next_release(parsed_args, version)
+    assert next == expected
+
+
+def test_get_no_current_release(no_tags_repo):
+    next = farmit.get_current_release(no_tags_repo)
+    assert next == None
```
Comment (ddriddle) [inline tests/test_farmit.py:189]: One-liner please:

```suggestion
    assert farmit.get_current_release(no_tags_repo) == None
```

**Review by ddriddle — DISMISSED** [review]
A bunch of cleanup items, and please finish the TODO item.

Clean up time

File: tests/test_farmit.py (hunk 4 above)
Comment (tzturner) [reply tests/test_farmit.py:59]: Seeking a bit of clarification on this item -- could you clarify how one might combine these two methods?

**Comment by tzturner** [comment]
@edthedev @ddriddle: @mpitcel and myself went through today and cleared every code request minus one, which we asked for clarification on.

File: tests/test_farmit.py (hunk 4 above)
Comment (mpitcel) [reply tests/test_farmit.py:60]: we should discuss this

Comment (edthedev) [reply tests/test_farmit.py:60]: Looks like we can if/then around lines 45 and 68, and keep most of the rest of the code the same.

Comment (edthedev) [reply tests/test_farmit.py:67]: Good catch. This is done.

**Review by mpitcel — APPROVED** [review]

**Comment by zdc217** [comment]
Closing for now. Will reopen if needed.

**Pull request closed without merging**